### PR TITLE
Allow imports without the lib or es6 directories

### LINKED ISSUE #43: Allow to import hyper-ts without es6 or lib directories
With the introduction of the `ReaderMiddleware` we need to choose between `es6` and `lib` directory for imports. As the rest of the `fp-ts` ecosystem support the import without these directories I think it is consequent to add it here too.

@DenisFrezzato do you have already something like this in the pipeline if not I can pick this up?

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Hopefully fixes #43 with a bunch of copy/pasting from the likes of https://github.com/gcanti/io-ts/pull/507.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,27 +2,27 @@
   "name": "hyper-ts",
   "version": "0.7.6",
   "description": "Type safe middleware architecture for HTTP servers",
-  "files": [
-    "lib",
-    "es6"
-  ],
   "main": "lib/index.js",
   "module": "es6/index.js",
   "typings": "lib/index.d.ts",
```
Comment (DenisFrezzato) [inline package.json:7]: I guess these must be updated too, right?

Comment (thewilkybarkid) [reply package.json:7]: This might be the same as https://github.com/gcanti/fp-ts/pull/1447, where it's not possible to install outside of npm directly.

Comment (DenisFrezzato) [reply package.json:7]: If there is no way to test it, I can release an `alpha` version with this change and see if it actually works.

Comment (DenisFrezzato) [reply package.json:7]: I've released 0.7.7-0, I've tried and it seems ok to me.

**Pull request merged**